### PR TITLE
[Forwardport] Add compare list link to success message after adding a product

### PR DESCRIPTION
**Pull request by gelanivishal**
### Original Pull Request
https://github.com/magento/magento2/pull/13862

### Description
If I add a product to the compare list, I want to go the compare list directly after that in many cases without having to search for a link. Thus, the link has been added to the success message now.

### Manual testing scenarios
1. Go to a product view page
2. Click "Add to compare"
3. Check if the success message contains a link to the compare list:
grafik

### Contribution checklist
 - [X] Pull request has a meaningful description of its purpose
 - [X] All commits are accompanied by meaningful commit messages
 - [X] All new or changed code is covered with unit/integration tests (if applicable)
 - [x] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by magento-engcom-team** [comment]
Hi @gelanivishal. Thank you for your contribution 
Here is some useful tips how you can test your changes using Magento test environment.
Add the comment under your pull request to deploy test or vanilla Magento instance:
- `@magento-engcom-team give me test instance` - deploy test instance based on PR changes
- `@magento-engcom-team give me {$VERSION} instance` - deploy vanilla Magento instance

For more details, please, review the [Magento Contributor Assistant](https://github.com/magento/magento2/wiki/Magento-Contributor-Assistant) documentation

**Comment by orlangur** [comment]
@gelanivishal don't forget to add Backport/Forwardport into issue title.

**Comment by nuzil** [comment]
@gelanivishal please fix Static and integration tests

File: app/code/Magento/Catalog/etc/frontend/di.xml [hunk 1]
```diff
@@ -79,23 +79,14 @@
             <argument name="typeId" xsi:type="string">recently_compared_product</argument>
         </arguments>
     </virtualType>
-    <type name="Magento\Catalog\Block\Product\View\Gallery">
+    <type name="Magento\Framework\View\Element\Message\MessageConfigurationsPool">
         <arguments>
-            <argument name="galleryImagesConfig" xsi:type="array">
```
Comment (nuzil) [inline app/code/Magento/Catalog/etc/frontend/di.xml]: Looks like you wrongly resolved conflicts while doing this Forwardport. Can you please check this part and resolve it correctly?

**Review by nuzil — CHANGES_REQUESTED** [review]

**Comment by gelanivishal** [comment]
@nuzil I have fixed request chagnes.

File: app/code/Magento/Catalog/etc/frontend/di.xml [hunk 2]
```diff
@@ -82,20 +82,11 @@
     <type name="Magento\Catalog\Block\Product\View\Gallery">
         <arguments>
             <argument name="galleryImagesConfig" xsi:type="array">
-                <item name="small_image" xsi:type="array">
```
Comment (nuzil) [inline app/code/Magento/Catalog/etc/frontend/di.xml]: Hi
Looks like you accidentally removed working part of code, can you please review your changes? As reference you can use this commit from original PR 
https://github.com/magento/magento2/pull/13862/files#diff-da4fdd9a985964091b2bbe0a3722ab77

**Review by nuzil — CHANGES_REQUESTED** [review]
@gelanivishal can you please fix failed tests?

**Comment by gelanivishal** [comment]
@nuzil Please review at once.

**Comment by gelanivishal** [comment]
@orlangur Please review requested changes.

**Comment by nuzil** [comment]
Hi @gelanivishal you still have failed Static tests, can you please review this changes?

**Comment by magento-engcom-team** [comment]
Hi @gelanivishal. Thank you for your contribution. 
We will aim to release these changes as part of 2.3.0. 
Please check the release notes for final confirmation.

**Pull request merged**